lib/artist.rb

```diff
@@ -0,0 +1,39 @@
+class Artist
+  extend Concerns::Findable
+
+  @@all = []
+
+  def initialize(name)
+    @name = name
+    @songs = []
+  end
+
+  attr_accessor :name
+  attr_accessor :songs
+
+  def self.all
+    @@all
+  end
+
+  def save
+    @@all << self
+    self
+  end
+
+  def self.destroy_all
+    @@all.clear
+  end
+
+  def self.create(name)
+    @@all.detect { |artist| artist.name == name } || Artist.new(name).save
```

Comment:
This method should simply **create** a new artist. The other functionality has already been handled by **find_or_create_by_name** in *Concerns::Findable*.